rnote-compose/src/shapes/arrow.rs

```diff
@@ -0,0 +1,212 @@
+    pub tip: na::Vector2<f64>,
+}
+
+impl TransformBehaviour for Arrow {
+    fn translate(&mut self, offset: na::Vector2<f64>) {
+        self.start += offset;
+        self.tip += offset;
+    }
+
+    fn rotate(&mut self, angle: f64, center: na::Point2<f64>) {
+        let isometry = {
+            let mut isometry = na::Isometry2::identity();
+            isometry.append_rotation_wrt_point_mut(&na::UnitComplex::new(angle), &center);
+            isometry
+        };
+
+        self.start = (isometry * na::Point2::from(self.start)).coords;
+        self.tip = (isometry * na::Point2::from(self.tip)).coords;
+    }
+
+    fn scale(&mut self, scale: na::Vector2<f64>) {
+        self.start = self.start.component_mul(&scale);
+        self.tip = self.tip.component_mul(&scale);
+    }
+}
+
+impl ShapeBehaviour for Arrow {
+    fn bounds(&self) -> Aabb {
+        let (x_values, y_values) = {
+            let lline = self.compute_lline(None);
+            let rline = self.compute_rline(None);
+
+            let x_values = [lline[0], rline[0], self.start[0], self.tip[0]];
+            let y_values = [lline[1], rline[1], self.start[1], self.tip[1]];
+
+            (x_values, y_values)
+        };
+
+        let bottom_left_corner = {
+            let lowest_x = x_values.into_iter().reduce(f64::min).unwrap();
+            let lowest_y = y_values.into_iter().reduce(f64::min).unwrap();
+            na::Point2::new(lowest_x, lowest_y)
+        };
+
+        let top_right_corner = {
+            let highest_x = x_values.into_iter().reduce(f64::max).unwrap();
+            let highest_y = y_values.into_iter().reduce(f64::max).unwrap();
+            na::Point2::new(highest_x, highest_y)
+        };
+
+        AabbHelpers::new_positive(bottom_left_corner, top_right_corner)
+    }
+
+    fn hitboxes(&self) -> Vec<Aabb> {
+        let n_splits = super::hitbox_elems_for_shape_len((self.tip - self.start).norm());
+
+        self.split(n_splits)
+            .into_iter()
+            .map(|line| line.bounds())
+            .collect()
+    }
+}
+
+/// Contains helper-functions for the `ShapeBehaviour` implementation of
+/// `Arrow`.
+impl Arrow {
+    /// Splits the stem of the arrow into the given number of rectangles.
+    pub fn split(&self, n_splits: i32) -> Vec<Line> {
+        (0..n_splits)
+            .map(|i| {
+                let sub_start = self
+                    .start
+                    .lerp(&self.tip, f64::from(i) / f64::from(n_splits));
+                let sub_end = self
+                    .start
+                    .lerp(&self.tip, f64::from(i + 1) / f64::from(n_splits));
+
+                Line {
+                    start: sub_start,
+                    end: sub_end,
+                }
+            })
+            .collect::<Vec<Line>>()
+    }
+}
+
+/// Functions which you can use to interact with the struct.
+impl Arrow {
+    /// The min-length for `rline` and `lline`.
+    const TIP_LINES_MIN_LENGTH: f64 = 32.0;
```

Comment:
I'd rename this to something like: `TIP_LINES_DEFAULT_LENGTH` and have a comment that says:

`The tip lines (rline & lline) default length, is the actual length when no stroke width is associated with the arrow,
if there is, it is used as the base length for the actual width calculation`.